src/System.Buffers.Primitives/System/Buffers/OwnedBuffer.cs

```diff
@@ -13,45 +12,17 @@ public abstract class OwnedBuffer<T> : IDisposable, IRetainable
 
         public abstract int Length { get; }
 
-        public abstract Span<T> Span { get; }
+        public abstract Span<T> AsSpan(int index, int length);
```

Comment:
>Does this mean that the IndexOf that takes an index/count vs Slice analysis doesn't apply to this? If so, and it is faster, then it is good to keep.

It does not apply because IndexOf and Slice on Span are non-virtual, can be inlined, and code optimized out.

>Why exactly can't the subtypes just slice the span?

Let's say the buffer wraps memory mapped file. AsSpan(0, 10) needs to read only first ten bytes of the file. Span.Slice(0, 10) would first have to read the whole file and then create a span of just first ten bytes.